src/plugins/intel_cpu/src/nodes/common/permute_kernel.cpp

```diff
@@ -146,121 +148,7 @@ struct jit_uni_permute_kernel_f32 : public jit_uni_permute_kernel, public jit_ge
-                                   new_dst_block_order[pos]);
-        new_dst_block_dims.insert(new_dst_block_dims.begin() + pos + 1,
-                                  params.src_block_dims[params.src_block_dims.size() - 1]);
-        new_dst_block_dims[pos] = div_up(new_dst_block_dims[pos], new_dst_block_dims[pos + 1]);
-        mask.insert(mask.begin() + pos + 1, 1);
-        mask[pos] = 1;
-    }
-
-    SizeVector sorted_src_strides;
-    SizeVector sorted_dst_strides;
-    SizeVector sorted_order;
-    SizeVector sorted_dst_dims;
-
-    //  support dynamic batch
-    int batch_ord = std::distance(params.order.begin(), std::find(params.order.begin(), params.order.end(), 0));
-    int batch_count = 0;
-    int batch_pos = 0;
-    for (size_t i = 0; i < new_dst_block_order.size(); i++) {
-        if (static_cast<int>(new_dst_block_order[i]) == batch_ord) {
-            batch_count++;
-            batch_pos = i;
-        }
-    }
-    if (batch_count == 1) {
-        sorted_src_strides.push_back(new_src_block_strides[batch_pos]);
-        sorted_dst_strides.push_back(new_dst_block_strides[batch_pos]);
-        sorted_order.push_back(new_dst_block_order[batch_pos]);
-        sorted_dst_dims.push_back(new_dst_block_dims[batch_pos]);
-        jcp.supported_dynamic_batch = true;
-    }
-
-    int n2 = 0;
-    for (size_t i = 0; i < mask.size(); i++) {
-        if (mask[i] == 0) {
-            n2++;
-            if (batch_count == 1 && static_cast<int>(new_dst_block_order[i]) == batch_ord) {
-                continue;
-            }
-            sorted_src_strides.push_back(new_src_block_strides[i]);
-            sorted_dst_strides.push_back(new_dst_block_strides[i]);
-            sorted_order.push_back(new_dst_block_order[i]);
-            sorted_dst_dims.push_back(new_dst_block_dims[i]);
-        }
-    }
-    for (size_t i = 0; i < mask.size(); i++) {
-        if (mask[i] == 1) {
-            sorted_src_strides.push_back(new_src_block_strides[i]);
-            sorted_dst_strides.push_back(new_dst_block_strides[i]);
-            sorted_order.push_back(new_dst_block_order[i]);
-            sorted_dst_dims.push_back(new_dst_block_dims[i]);
-        }
-    }
-
-    int max_threads = parallel_get_max_threads();
-    const int n_max = 3;    //  max count dims for parallel
-    int n = 0;
-    int work_amount = sorted_dst_dims[0];
-    for (size_t i = 1; i < sorted_dst_dims.size() && n < n_max; i++) {
-        n++;
-        if (work_amount >= 4 * max_threads) {   //  4 * max_threads is a specially selected value for best performance
-            break;
-        }
-        work_amount *= sorted_dst_dims[i];
-    }
-
-    jcp.src_strides = sorted_src_strides;
-    jcp.dst_strides = sorted_dst_strides;
-    jcp.dst_block_dims = sorted_dst_dims;
-    jcp.n = std::min(n, n2);
-    jcp.ndims = sorted_order.size();
-    jcp.data_size = params.data_size;
-
+    jcp = TransposeExecutor::prepareParams(params);
```

Comment:
Just curious. Logically `TransposeExecutor` should call some methods of PermuteKernel, so that it depends on some less comlex structure and not vise versa. What is the need to make PermuteKernel dependent on the TransposeExecutor?

Reply:
The idea is to remove `PermuteKernel ` at all, its logic will be encapsulated into jit and ref executors.
However, several ops depend on `PermuteKernel `, so it was decided to keep it temporary and use as a "proxy" to executor logic.

Reply:
Ok. Now it's clear. Hope we will refactor this soon.